framework/src/utils/FormattedTable.C

```diff
@@ -112,7 +112,8 @@ void
 FormattedTable::addData(const std::string & name, Real value, Real time)
 {
   _data[time][name] = value;
-  _column_names.insert(name);
+  if (std::find(_column_names.begin(), _column_names.end(), name) == _column_names.end())
```

Comment:
We call this method at the end of every timestep a few times (1-20). It's not super hot and might actually be faster for small numbers of PPs, but we might want to watch it.

There are several ways we could fix it. If we wanted to do it internally, we could build an index cache every time a new column is added where we lookup the index and insert into the right spot. If we wanted to change the API we could return the index to the user and create a second method where they could do an insert based on an index instead of doing this string lookup every time. I actually like that idea quite a bit. 

I would say that it's not necessary as part of this PR though as it'll mean you'll probably have to touch the Outputs system too.